awscli/customizations/s3/s3handler.py

```diff
@@ -65,14 +68,16 @@ def __init__(self, session, params, result_queue=None,
         for key in self.params.keys():
             if key in params:
                 self.params[key] = params[key]
-        self.multi_threshold = multi_threshold
-        self.chunksize = chunksize
+        self.multi_threshold = self._runtime_config['multipart_threshold']
+        self.chunksize = self._runtime_config['multipart_chunksize']
+        LOGGER.debug("Using a multipart threshold of %s and a part size of %s",
+                     self.multi_threshold, self.chunksize)
         self.executor = Executor(
-            num_threads=self.EXECUTOR_NUM_THREADS,
+            num_threads=self._runtime_config['max_concurrent_requests'],
```

Comment:
After initially implementing that, I ended up making this just match the number of threads.  The main reason is that if we do add ` max - 1`, then the minimum value needs to be `2`, which I think would confuse customers.  It's also not intuitive that you can't set this value to 1.  If a user wants to "turn off" parallelism, you'd think you'd set `max_concurrent_requests` to 1, but they would need to set max_concurrent_requests to `2`.  I could see us getting questions about that.

Another option is to rename this to be more explicit.  If we called it `max_transfer_requests` maybe that would clear up what this is doing.  Although I think `max_concurrent_requests` is more clear than `max_transfer_threads`.  What do you think?